src/components/Modal/ExtensionStatusModal.tsx

```diff
@@ -0,0 +1,203 @@
+            : styles.extensionPending;
+
+    if (!isOpen) return null;
+
+    if (isLoading) {
+        return (
+            <div className={styles.extensionModalOverlay}>
+                <div className={styles.extensionModalLoading}>
+                    <h2>Extension Details</h2>
+                    <div className={styles.spinnerContainer}>
+                        <SmallSpinner />
+                    </div>
+                </div>
+            </div>
+        );
+    }
+
+    if (error) {
+        return (
+            <div className={styles.extensionModalOverlay}>
+                <div className={styles.extensionModal}>
+                    <h2>Error loading extension requests</h2>
+                    <pre>{JSON.stringify(error, null, 2)}</pre>
+                    <button
+                        className={styles.extensionCloseButton}
+                        onClick={onClose}
+                    >
+                        Close
+                    </button>
+                </div>
+            </div>
+        );
+    }
+
+    const extensionRequests = data?.allExtensionRequests ?? [];
+    const hasPendingRequest = extensionRequests.some(
+        (req) => req.status === 'PENDING'
+    );
+
+    return (
+        <div
+            className={styles.extensionModalOverlay}
+            onClick={handleOutsideClick}
+        >
+            <div className={styles.extensionModal} ref={modalRef}>
+                <h2>Extension Details</h2>
+
+                {extensionRequests.length > 0 ? (
+                    extensionRequests.map((request) => (
+                        <div
+                            key={request.id}
+                            className={styles.extensionExtensionRequest}
+                        >
+                            {[
+                                {
+                                    label: 'Request :',
+                                    value: `#${request.requestNumber}`,
+                                },
+                                {
+                                    label: 'Reason :',
+                                    value: request.reason,
+                                },
+                                { label: 'Title :', value: request.title },
+                                {
+                                    label: 'Old Ends On :',
+                                    value: formatDate(request.oldEndsOn),
+                                },
+                                {
+                                    label: 'New Ends On :',
+                                    value: formatDate(request.newEndsOn),
+                                },
+                                {
+                                    label: 'Status :',
+                                    value: request.status,
+                                    className: getStatusClass(request.status),
+                                },
+                            ].map((item, idx) => (
```

Comment:
- insetad of defining the array here, making it difficult to read what's happening, can we create a function that returns a value which can be used here? as a side effect later if we want to memorize the value for any reason we can do that aswell.